sdk/client/src/verifying_client.rs

```diff
@@ -0,0 +1,516 @@
+        let (latest_li, epoch_change_proof, _) = state_proof;
+
+        let trusted_state = self.trusted_state();
+        let change = trusted_state.verify_and_ratchet(latest_li, epoch_change_proof)?;
+
+        if let Some(new_state) = change.new_state() {
+            self.trusted_state_store
+                .write()
+                .unwrap()
+                .ratchet(new_state)?;
+        }
+
+        // TODO(philiphayes): verify the accumulator consistency proof; no-one
+        // seems to actually verify this anywhere?
+
+        Ok(())
+    }
+
+    pub async fn get_account(
+        &self,
+        address: AccountAddress,
+    ) -> Result<Response<Option<AccountView>>> {
+        Ok(self
+            .send_via_batch(MethodRequest::GetAccount((address,)))
+            .await?
+            .and_then(MethodResponse::try_into_get_account)?)
+    }
+
+    /// Send a single request via `VerifyingClient::batch`.
+    async fn send_via_batch(&self, request: MethodRequest) -> Result<Response<MethodResponse>> {
+        let mut responses = self.batch(vec![request]).await?.into_iter();
+        let response = match responses.next() {
+            Some(response) => response,
+            None => bail!("response batch is empty"),
+        };
+        ensure!(responses.next().is_none(), "too many responses");
+        response
+    }
+
+    pub async fn batch(
+        &self,
+        requests: Vec<MethodRequest>,
+    ) -> Result<Vec<Result<Response<MethodResponse>>>> {
+        // transform each request into verifying sub-request batches
+        let batch = VerifyingBatch::from_batch(requests);
+
+        // flatten and collect sub-request batches into flat list of requests
+        let mut requests = batch.collect_requests();
+        // append get_state_proof request
+        let start_version = self.version();
+        requests.push(MethodRequest::get_state_proof(start_version));
+
+        // actually send the batch
+        let responses = self.inner.batch(requests).await?;
+
+        // TODO(philiphayes): REMOVE. for now, just convert the errors until we
+        // use the real error type.
+        let mut responses = responses
+            .into_iter()
+            .map(|result| result.map_err(anyhow::Error::new))
+            .collect::<Vec<_>>();
+
+        // extract and verify the state proof
+        let (state_proof_response, state) = responses.pop().unwrap()?.into_parts();
+        let state_proof_view = state_proof_response.try_into_get_state_proof()?;
+        let state_proof: StateProof = (&state_proof_view).try_into()?;
+
+        // check the response metadata matches the state proof
+        let latest_li = state_proof.0.ledger_info();
+        ensure!(state.version == latest_li.version());
+        ensure!(state.timestamp_usecs == latest_li.timestamp_usecs());
+
+        // try to ratchet our trusted state using the state proof
+        self.verify_and_ratchet(&state_proof)?;
```

Comment:
Should call `sync_one_step()`?

Reply:
Going to address this in a follow up I think